src/msg/Message.cc

```diff
@@ -255,47 +255,45 @@ void Message::encode(uint64_t features, int crcflags, bool skip_header_crc)
 
   if (crcflags & MSG_CRC_DATA) {
     calc_data_crc();
-
-#ifdef ENCODE_DUMP
-    bufferlist bl;
-    encode(get_header(), bl);
-
-    // dump the old footer format
-    ceph_msg_footer_old old_footer;
-    old_footer.front_crc = footer.front_crc;
-    old_footer.middle_crc = footer.middle_crc;
-    old_footer.data_crc = footer.data_crc;
-    old_footer.flags = footer.flags;
-    encode(old_footer, bl);
-
-    encode(get_payload(), bl);
-    encode(get_middle(), bl);
-    encode(get_data(), bl);
-
-    // this is almost an exponential backoff, except because we count
-    // bits we tend to sample things we encode later, which should be
-    // more representative.
-    static int i = 0;
-    i++;
-    int bits = 0;
-    for (unsigned t = i; t; bits++)
-      t &= t - 1;
-    if (bits <= 2) {
-      char fn[200];
-      int status;
-      snprintf(fn, sizeof(fn), ENCODE_STRINGIFY(ENCODE_DUMP) "/%s__%d.%x",
-	       abi::__cxa_demangle(typeid(*this).name(), 0, 0, &status),
-	       getpid(), i++);
-      int fd = ::open(fn, O_WRONLY|O_TRUNC|O_CREAT|O_CLOEXEC|O_BINARY, 0644);
-      if (fd >= 0) {
-	bl.write_fd(fd);
-	::close(fd);
-      }
-    }
-#endif
   } else {
     footer.flags = (unsigned)footer.flags | CEPH_MSG_FOOTER_NOCRC;
   }
+  if (header.type == MSG_PGSTATS) {
+    if (cct) {
+      ldout(cct, 0) << "We are encoding MSG_PGSTATS; "
```

Comment:
This might be too verbose. Having it at `0` strips the flexibility of turning this (potentially costly) printing off without upgrade. I would put the debug at higher level.